### PR TITLE
chore(core): update 000_null_keyboard to be a proper null keyboard 🙀

### PR DESCRIPTION
**Pull request by srl295**
- rewrite 000_null_keyboard.kmx to be valid, but with no keys
- remove outdated comment in kmx_plus.h
- update kmx_plus.cpp to not reverse the order of section ids (#7098)
- add a new test option for expected-error

@keymanapp-test-bot skip

for #7097 
as always, #5015

### CODE REVIEW

File: core/src/ldml/ldml_processor.cpp [hunk 1]
```diff
@@ -188,7 +188,6 @@ ldml_processor::process_event(
       assert(keys->header.ident == LDML_SECTIONID_KEYS);
       // Look up the key
       const kmx::COMP_KMXPLUS_KEYS_ENTRY *key =  keys->find(vk, modifier_state);
-      assert(key != nullptr);
```
Comment (srl295) [inline core/src/ldml/ldml_processor.cpp:191]: not a runtime err

File: core/src/kmx/kmx_plus.h [hunk 2]
```diff
@@ -154,7 +154,6 @@ validate_as_section(const uint8_t *data, uint32_t ident) {
     return nullptr;
   }
   const COMP_KMXPLUS_HEADER *all = reinterpret_cast<const COMP_KMXPLUS_HEADER *>(data);
-  // TODO-LDML these fail on 000null ..
```
Comment (srl295) [inline core/src/kmx/kmx_plus.h:157]: no longer fails!

**Review by mcdurdin — APPROVED** [review]

**Review by rc-swag — APPROVED** [review]
lgtm

**Comment by srl295** [comment]
will merge, #7124 adds validation and addresses comments here

**Pull request merged**